### PR TITLE
Create appointment step 3

### PR DESCRIPTION
**Pull request by LomyW**
Ui to create an appointment with optional weekly recurring appointments.

### CODE REVIEW

File: src/components/DatePicker.tsx [hunk 1]
```diff
@@ -16,7 +18,15 @@ const DatePicker: React.FC<Props> = ({ type = 'date', onChange, value, useMin =
     return (
         <>
             <div className="lf__datepicker">
-                <input type={type} onChange={onChange} value={value} min={(type === 'date' && useMin && _min) || undefined} />
+                <input
+                    placeholder={type === 'date' ? 'Wähle ein Datum...' : '00:00'}
```
Comment (timidak) [inline src/components/DatePicker.tsx:22]: i18n should be used here

File: src/components/appointment/AppointmentDetail.tsx [hunk 2]
```diff
@@ -19,20 +19,27 @@ const AppointmentDetail: React.FC<AppointmentDetailProps> = ({ appointment, cour
...
     const [canceled, setCanceled] = useState<boolean>(false);
+    const { date, startTime, endTime } = getAppointmentDateTime(appointment.start, appointment.duration);
+
+    const countAttendees = useCallback(() => {
+        // TODO change to appointment.appointment_participant_pupil, appointment.appointment_participant_student, appointment.appointment_participant_screener
+        const participants = appointment.participants ? appointment.participants.length : 0;
+        const organizers = appointment.organizers ? appointment.organizers.length : 0;
+        return participants + organizers;
+    }, [appointment.organizers, appointment.participants]);
+    const attendeesCount = useMemo(() => countAttendees(), [countAttendees]);
```
Comment (timidak) [inline src/components/appointment/AppointmentDetail.tsx:30]: I guess these are the right dependencies
```suggestion
    const attendeesCount = useMemo(() => countAttendees(), [appointment.organizers, appointment.participants]);
```

File: src/components/appointment/Description.tsx [hunk 3]
```diff
@@ -15,12 +15,9 @@ const Description: React.FC<AvatarsProps> = ({ appointmentType, courseName, cour
             <Divider thickness="0.25" my={5} />
             <VStack p={3}>
                 <Text color="primary.900" mb="2">
-                    {t(
-                        appointmentType === AppointmentTypes.GROUP
-                            ? 'appointment.appointmentDetail.courseDescriptionHeader'
-                            : 'appointment.appointmentDetail.desciptionHeader',
-                        { courseTitle: courseName }
-                    )}
+                    {t(appointmentType === AppointmentTypes.GROUP ? 'appointment.detail.courseDescriptionHeader' : 'appointment.detail.desciptionHeader', {
```
Comment (timidak) [inline src/components/appointment/Description.tsx:18]: typo: 'appointment.detail.desciptionHeader' -> 'appointment.detail.desc**r**iptionHeader'

File: src/components/appointment/Description.tsx [hunk 4]
```diff
@@ -15,12 +15,9 @@ const Description: React.FC<AvatarsProps> = ({ appointmentType, courseName, cour
...
+                        courseTitle: courseName,
+                    })}
                 </Text>
                 <Text color="primary.600" fontWeight="normal">
                     {appointmentType === AppointmentTypes.GROUP ? courseDescription : 'Hier kommt die appointment description.'}
```
Comment (timidak) [inline src/components/appointment/Description.tsx:23]: 'Hier kommt die appointment description.' ? Is this the final text? 😄 
Add to i18n

File: src/context/AppointmentContext.tsx [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    CreateAppointmentAction,
+    FormReducerActionType,
+    State,
+    Weeklies,
+    WeeklyAppointmentAction,
+    WeeklyReducerActionType,
+    TAppointmentContext,
+} from '../types/lernfair/CreateAppointment';
+
+const formReducer: Reducer<State, CreateAppointmentAction> = (state: State, action: CreateAppointmentAction) => {
```
Comment (timidak) [inline src/context/AppointmentContext.tsx:12]: createAppointmentFormReducer? ☺️

File: src/pages/create-appointment/AppointmentCreation.tsx [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+    const validateInputs = () => {
+        const isDateMinOneWeekLater = (date: string): boolean => {
+            const startDate = DateTime.fromISO(date);
+            const diff = startDate.diffNow('days').days;
+            if (diff >= 7) return true;
+            return false;
+        };
+
+        if (!appointmentToCreate.title.length) {
+            setErrors({ ...errors, title: 'Title darf nicht leer sein' });
```
Comment (timidak) [inline src/pages/create-appointment/AppointmentCreation.tsx:84]: use the i18n constants you already added here

File: src/pages/create-appointment/AppointmentCreation.tsx [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+            return false;
+        } else {
+            delete errors.title;
+        }
+        if (!appointmentToCreate.date) {
+            setErrors({ ...errors, date: 'Datum darf nicht leer sein' });
```
Comment (timidak) [inline src/pages/create-appointment/AppointmentCreation.tsx:90]: > use the i18n constants you already added here

same here

File: src/pages/create-appointment/AppointmentCreation.tsx [hunk 8]
```diff
@@ -0,0 +1,202 @@
...
+            delete errors.date;
+        }
+        if (isDateMinOneWeekLater(appointmentToCreate.date) === false) {
+            setErrors({ ...errors, dateNotInOneWeek: 'Datum muss in mind einer Woche sein' });
+            return false;
+        } else {
+            delete errors.date;
+        }
+        if (!appointmentToCreate.time.length) {
+            setErrors({ ...errors, time: 'Zeit darf nicht leer sein' });
```
Comment (timidak) [inline src/pages/create-appointment/AppointmentCreation.tsx:102]: > use the i18n constants you already added here

same here

File: src/pages/create-appointment/AppointmentCreation.tsx [hunk 9]
```diff
@@ -0,0 +1,202 @@
...
+            return false;
+        } else {
+            delete errors.time;
+        }
+        if (appointmentToCreate.duration === 0) {
+            setErrors({ ...errors, duration: 'Dauer darf nicht leer sein' });
```
Comment (timidak) [inline src/pages/create-appointment/AppointmentCreation.tsx:108]: > use the i18n constants you already added here

same here

File: src/pages/create-appointment/AppointmentCreation.tsx [hunk 10]
```diff
@@ -0,0 +1,202 @@
...
+            delete errors.title;
+        }
+        if (!appointmentToCreate.date) {
+            setErrors({ ...errors, date: 'Datum darf nicht leer sein' });
+            return false;
+        } else {
+            delete errors.date;
+        }
+        if (isDateMinOneWeekLater(appointmentToCreate.date) === false) {
+            setErrors({ ...errors, dateNotInOneWeek: 'Datum muss in mind einer Woche sein' });
```
Comment (timidak) [inline src/pages/create-appointment/AppointmentCreation.tsx:96]: Datum muss in mind einer Woche sein -> Datum kann frühestens in einer Woche sein

Comment (timidak) [reply src/pages/create-appointment/AppointmentCreation.tsx:96]: and add to i18n ☺️

**Review by timidak — CHANGES_REQUESTED** [review]
A few changes, rest looks pretty good to me!

File: src/components/appointment/Description.tsx (hunk 4 above)
Comment (LomyW) [reply src/components/appointment/Description.tsx:23]: will treat in another PR, because there are some more changes to the appointment details

**Review by timidak — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**